examples/shader/storage_buffer.rs

```diff
@@ -0,0 +1,113 @@
+//! This example demonstrates how to use a storage buffer with `AsBindGroup` in a custom material.
```

Comment:
Should the example be renamed to match the new type? I don't really have an opinion either way honestly.

Reply:
My thought was just that `shader/shader_storage_buffer` was a bit redundant.

Reply:
But happy to change :)